cluster/watchdog.go

```diff
@@ -107,7 +107,7 @@ func (w *Watchdog) rescheduleContainers(e *Engine) {
 			clusterNetworks := w.cluster.Networks().Uniq()
 			for networkName, endpoint := range c.Info.NetworkSettings.Networks {
 				net := clusterNetworks.Get(endpoint.NetworkID)
-				if net != nil && net.Scope == "global" {
+				if net != nil && (net.Scope == "global" || net.Scope == "swarm") {
```

Comment:
When running Swarm with Swarm mode enabled, we need to disconnect and reconnect swarm-scoped networks as well, not just global-scoped ones. swarm-scoped networks are, in effect, globally-scoped ones.